newlib/libc/sys/vita/vitanet.c

```diff
@@ -0,0 +1,93 @@
+/*
+
+Copyright (C) 2022, vitasdk
+
+Permission is hereby granted, free of charge, to any person obtaining a
+copy of this software and associated documentation files (the "Software"),
+to deal in the Software without restriction, including without limitation
+the rights to use, copy, modify, merge, publish, distribute, sublicense,
+and/or sell copies of the Software, and to permit persons to whom the
+Software is furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
+THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+DEALINGS IN THE SOFTWARE.
+
+*/
+
+#include <psp2/net/net.h>
+#include <psp2/net/netctl.h>
+#include <psp2/sysmodule.h>
+#include <psp2/types.h>
+#include <stdlib.h>
+
+#include "vitanet.h"
+
+// theoretical maximum:
+// 128 bytes per socket, 1024 sockets max
+// 4KiB reserved
+// 3KiB - basic features
+// 4KiB - dns cache
+// 1300 bytes - dns resolver
+#define NET_MEM_SIZE (128*1024 + ((4+3+4) * 1024)) + 1300
+
+#ifndef SCE_NET_CTL_ERROR_NOT_TERMINATED
+#define SCE_NET_CTL_ERROR_NOT_TERMINATED 0x80412102
+#endif
+
+static void *_vita_net_memory = NULL;
+
+int _vita_net_init()
+{
+	int ret;
+	SceNetInitParam net_init_param = {0};
```

Comment:
maybe this approach is useful. but most projects already have network initialization code in their codeset.
also it makes useless calling when we use the network functions in every times.
so I'm worried it can make the perf problem to homebrews.

how about add a flag variable in top of the function to bypass below logics?

for example
```
static bool _VITA_NET_INITIALIZED = false;

int _vita_net_init()
{
  if (_VITA_NET_INITIALIZED)
    return 0;
  ...
  ret = secNetInit(...);
  ...
  if (ret != SCE_NET_ERROR_EBUSY)
    goto success;
  ...
success:
  _VITA_NET_... = true;
  return 0;
}
```